randgen: fix postgres mutator for session vars

rafiss opened the pull request:
fixes https://github.com/cockroachdb/cockroach/issues/67791

Release note: None

rafiss committed:
randgen: fix postgres mutator for session vars

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/82548)
<!-- Reviewable:end -->

otan commented on pkg/sql/randgen/mutator.go:
```diff
@@ -832,6 +833,8 @@ func postgresCreateTableMutator(
 			}
 			stmt.Defs = newdefs
+		case *tree.SetVar:
+
```
hmm, does this mean we lose out on stuff like `SET intervalstyle = '...'`, and do we (deliberately) want to do that?

rafiss committed:
compare_test: add extensions to Postgres setup

Release note: None

rafiss committed:
sql: don't error for fuzzystrmatch and pgcrypto extensions

We support enough of these extensions that we should start allowing
these statements, similar to how we did for pg_trgm.

Release note: None

rafiss commented:
bors r=otan
